config/initializers/0_application_config.rb

```diff
@@ -1,10 +1,8 @@
-ENVied.require(*ENV["ENVIED_GROUPS"] || Rails.groups)
-
 class ApplicationConfig
   URI_REGEXP = %r{(?<scheme>https?://)?(?<host>.+?)(?<port>:\d+)?$}.freeze
 
   def self.[](key)
-    ENVied.public_send(key)
+    ENV[key]
```

Comment:
**suggestion (non-blocking):** Should we use `ENV.fetch(key)` here? The previous code raises a `NameError` for invalid keys, the current one returns `nil` instead. Changing to `fetch` would raise a `KeyError` instead.

Reply:
One of the complaints I have about ENVied is that the entire app crashes if an ENV variable is missing when in reality letting it be nil might not be a huge deal which makes me want to let it be nil here rather than raise an error. Not married to that idea but that's why I did it this way

Reply:
The main problem I had when using ENV directly, were hard to track down bugs due to the fact that there's no indication if the value is missing on purpose or due to e.g. a typo, especially when combined with `.present?` checks:

```ruby
ENV['RAILS_LOG_TO_STDOUT'].present?
#=> true
ENV['RAILS_LOG_TO_STODUT'].present?
#=> false
```

A typo like the second example can go unnoticed for quite a while in my experience.

Maybe a sort of compromise could work:

```ruby
def self.[](key)
  if ENV.has_key?(key)
    ENV[key]
  else
    # Record missing access in log, DD, or wherever
    Rails.logger.warn("Unset ENV variable: #{key}.")
  end
end
```

This doesn't crash the app but at least leaves a message when trying to access a missing value, so we can perform a  sanity check. Alternatively, this could be achieved with a sentinel value, which at least distinguishes it from an accidental `nil`:

```ruby
def self.[](key)
  ENV.fetch(key, :missing_variable)
end

ApplicationConfig["RAILS_LOG_TO_STODUT"]
#=> :missing_variable
```

Reply:
LOVE the logging idea and then we could set up a monitor in Datadog to detect those logs so we get alerted but the whole app won't crash.

Reply:
That sounds great @mstruve 🎉